cpp/src/parquet/column_reader.h

```diff
@@ -115,6 +116,17 @@ class PARQUET_EXPORT PageReader {
                                           bool always_compressed = false,
                                           const CryptoContext* ctx = NULLPTR);
 
+  // @returns: true if the skip callback was successfully set. Returns false
+  // if the callback was not set or not supported.
+  // If supported, NextPage() will use this callback to determine if it should
```

Comment:
Are there contexts where this wouldn't be supported? Why?

Reply:
I do not have a concrete example. Since this is an abstract class, I tried to make it a more general case.

Reply:
if we don't have a failure case here in mind, then lets either make the method void (YAGNI) if we think there is a possible error, then Status should be returned.  Similar question is does this need to be virtual?

Reply:
Done.